### ISSUE TITLE
This seems wrong

### ISSUE THREAD

**Issue by janz1961** [timmcmic/DLConversionV2#48]
Start-DistributionListMigration kept crashing and then I found:

Set-Office365DL.ps1:59   [boolean]$functionReportToOriginator=$NULL

Assigning $NULL to a boolean? Nah.  On my computer this resulted in an exception. I changed it to $false. I'm assuming that PowerShell used to treat $NULL as false (just as they do zero) and this is why this problem never surfaced before. 

I'm not really experienced with collaborative development and don't know how I could submit this change. Or if changes by me would even be accepted. But, I can let you know it works now. I can finally migrate thos DL's.

**Comment by timmcmic** [comment]
Are you using powershell 7 or 5.x


==============================
Timothy J. McMichael
Senior Support Escalation Engineer
***@***.******@***.***>
(980)-776-7465

Hours:  Sunday – Wednesday 08:00 – 16:00 eastern time zone.

Manager:  Tom Roughley ***@***.******@***.***>)

Premier Support - (800)-936-3100
Broad Commercial Support - (800)-936-4900

==============================

________________________________
From: janz1961 ***@***.***>
Sent: Thursday, April 28, 2022 1:47 PM
To: timmcmic/DLConversionV2 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [timmcmic/DLConversionV2] This seems wrong (Issue #48)


Start-DistributionListMigration kept crashing and then I found:

Set-Office365DL.ps1:59 [boolean]$functionReportToOriginator=$NULL

Assigning $NULL to a boolean? Nah. On my computer this resulted in an exception. I changed it to $false. I'm assuming that PowerShell used to treat $NULL as false (just as they do zero) and this is why this problem never surfaced before.

I'm not really experienced with collaborative development and don't know how I could submit this change. Or if changes by me would even be accepted. But, I can let you know it works now. I can finally migrate thos DL's.

**Comment by janz1961** [comment]
See below. Added the easiest way to reproduce the error.

```
PS C:\> get-host | Select-Object Version

Version
-------
5.1.14393.5066

PS C:\> [boolean]$functionReportToOriginator=$NULL
Cannot convert value "" to type "System.Boolean". Boolean parameters accept only Boolean values and numbers, such as $True,
$False, 1 or 0.
At line:1 char:1
+ [boolean]$functionReportToOriginator=$NULL
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : MetadataError: (:) [], ArgumentTransformationMetadataException
    + FullyQualifiedErrorId : RuntimeException

```

**Comment by timmcmic** [comment]
Cool!  It’s wrong anyway but I was just curious why it was happening all of a sudden.  I’ll get it fixed Sunday and post a new version.


==============================
Timothy J. McMichael
Senior Support Escalation Engineer
***@***.******@***.***>
(980)-776-7465

Hours:  Sunday – Wednesday 08:00 – 16:00 eastern time zone.

Manager:  Tom Roughley ***@***.******@***.***>)

Premier Support - (800)-936-3100
Broad Commercial Support - (800)-936-4900

==============================

________________________________

**Comment by janz1961** [comment]
Yeah, I didn’t imagine that such a comprehensive set of tooling was only use by me because everyone should have ran into it 😊

Maybe you can solve another puzzle for me. While debugging it frustrated the hell out of me that every time the commend exited the shell (and in case of the ISE, the ISE). Why is this and is there a way around it. My workaround was using the ISE and keeping a modified file in it (so it would ask if I wanted to save it). But this was not ideal.

Jan

**Comment by timmcmic** [comment]
So – the exit is by design to support multi-threading and job exiting.

What you can do is this.

Open a command prompt (not powershell).

Type powershell.exe.

This opens powershell from within command.

When the exit is called it exits to command.

When you run powershell.exe again it remembers the commands you previously ran by up arrowing.

DO not use ise.

Tim

==============================
Timothy J. McMichael
Senior Support Escalation Engineer
***@***.******@***.***>
(980)-776-7465

Hours:  Sunday – Wednesday 08:00 – 16:00 eastern time zone.

Manager:  Tom Roughley ***@***.******@***.***>)

Premier Support - (800)-936-3100
Broad Commercial Support - (800)-936-4900

==============================

**Comment by timmcmic** [comment]
Fixed 2.5.20 and published today.
